### PR TITLE
add share functionality

### PR DESCRIPTION
**Pull request by Artistan**
## reference
https://docs.gitlab.com/ee/api/projects.html#share-project-with-group

### CODE REVIEW

File: lib/Gitlab/Api/Projects.php [hunk 1]
```diff
@@ -665,4 +665,24 @@ public function deployment($project_id, $deployment_id)
...
     }
+    
+    /**
+     * @param mixed $project_id
+     * @param array $params
+     * @return mixed
+     */
+    public function addShare($project_id, array $params = array())
+    {
+        return $this->post($this->getProjectPath($project_id, 'share'), $params);
```
Comment (m1guelpf) [inline lib/Gitlab/Api/Projects.php]: Can you use our options resolver for the `$params` array?

**Comment by Artistan** [comment]
@m1guelpf - updated with resolver.

**Comment by m1guelpf** [comment]
@Artistan Could you also add support for the `expires_at` parameter?

**Comment by Artistan** [comment]
@m1guelpf - updated with required and expires_at

It looks like a lot of the normalizers are broken if my test is any indication.
http://symfony.com/doc/current/components/options_resolver.html#option-normalization
a lot of the normalizers I saw do not have the "Option" as the first parameter

**Pull request merged**